src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs

```diff
@@ -0,0 +1,204 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Nethermind.Core.Extensions;
+using Nethermind.Evm;
+using Nethermind.Logging;
+
+namespace Nethermind.Evm.EOF;
+
+public interface IEofVersionHandler
+{
+    bool ValidateCode(ReadOnlySpan<byte> code);
+    bool TryParseEofHeader(ReadOnlySpan<byte> code, out EofHeader? header);
+}
+
+public class EvmObjectFormat
+{
+    // magic prefix : EofFormatByte is the first byte, EofFormatDiff is chosen to diff from previously rejected contract according to EIP3541
+    private static byte[] EOF_MAGIC = { 0xEF, 0x00 };
+
+    private readonly Dictionary<byte, IEofVersionHandler> _eofVersionHandlers = new();
+
+    private readonly ILogger _logger;
+    private bool LoggingEnabled => _logger?.IsTrace ?? false;
+    public EvmObjectFormat(ILogManager logManager = null)
+    {
+        _logger = logManager?.GetClassLogger<EvmObjectFormat>();
+        _eofVersionHandlers.Add(0x01, new Eof1(logManager));
+    }
+
+    /// <summary>
+    /// returns whether the code passed is supposed to be treated as Eof regardless of its validity.
+    /// </summary>
+    /// <param name="container">Machine code to be checked</param>
+    /// <returns></returns>
+    public bool IsEof(ReadOnlySpan<byte> container) => container.StartsWith(EOF_MAGIC);
+
+    public bool IsValidEof(ReadOnlySpan<byte> container)
+    {
+        if (container.Length < 7)
```

Comment:
Shouldn't this call `IsEof(ReadOnlySpan<byte> container)` to check for the prefix ?

Reply:
We dont call this unless it is Eof afaik